src/lua/api-gateway/validation/validator.lua

```diff
@@ -147,6 +147,7 @@ function BaseValidator:setKeyInRedis(key, hash_name, keyexpires, value)
     local rediss = redis:new()
     local redis_host, redis_port = self:getRedisUpstream(redis_RW_upstream)
     local ok, err = rediss:connect(redis_host, redis_port)
+    rediss:set_timeout(5000)
```

Comment:
I would have this as part of another 'pull-request'. Just reducing the timeout only helps failing the requests faster, but what we actually want is to have errors while caching the token not affecting the API GW serving the request.